streams/writable-streams/aborting.any.js

```diff
@@ -1376,3 +1376,99 @@ promise_test(t => {
   return promise_rejects_js(t, TypeError, ws.abort(), 'abort should reject')
     .then(() => writer.ready);
 }, 'abort on a locked stream should reject');
+
+test(t => {
+  let ctrl;
+  const ws = new WritableStream({start(c) { ctrl = c; }});
+
+  assert_true(ctrl.signal instanceof AbortSignal);
+  assert_false(ctrl.signal.aborted);
+  ws.abort();
+  assert_true(ctrl.signal.aborted);
+}, 'WritableStreamDefaultController.signal');
+
+promise_test(async t => {
+  let ctrl;
+  let resolve;
+  const called = new Promise(r => resolve = r);
+
+  const ws = new WritableStream({
+    start(c) { ctrl = c; },
+    write() { resolve(); return new Promise(() => {}); }
+  });
+
+  writer.write(99);
+  await called;
+
+  assert_false(ctrl.signal.aborted);
+  ws.abort();
+  assert_true(ctrl.signal.aborted);
+}, 'the abort signal is signalled synchronously - write');
+
+promise_test(async t => {
+  let ctrl;
+  let resolve;
+  const called = new Promise(r => resolve = r);
+
+  const ws = new WritableStream({
+    start(c) { ctrl = c; },
+    close() { resolve(); return new Promise(() => {}); }
+  });
+
+  writer.close(99);
+  await called;
+
+  assert_false(ctrl.signal.aborted);
+  ws.abort();
+  assert_true(ctrl.signal.aborted);
+}, 'the abort signal is signalled synchronously - close');
+
+promise_test(async t => {
+  let ctrl;
+  const ws = new WritableStream({start(c) { ctrl = c; }});
+  const writer = ws.getWriter();
+
+  ctrl.error(TypeError());
+  await promise_rejects_js(t, TypeError, writer.closed);
```

Comment:
This should probably use `promise_rejects_exactly` and use a specific error object like `error1` or `error2`. That way, we won't get a false positive if an incorrect implementation rejects with a *different* `TypeError`.

(Same thing for the other tests.)

Reply:
Done.